src/app/store/effects/router.effect.ts

```diff
@@ -22,8 +22,10 @@ export class RouterEffects {
   navigate$ = this.actions$.pipe(
     ofType(RouterActions.GO),
     map((action: RouterActions.Go) => action.payload),
-    tap(({ path, query: queryParams, extras }) => {
-      this.router.navigate(path, { queryParams, ...extras });
+    tap(({ path, query: queryParams, extras, callback, errorHandler }) => {
```

Comment:
have learned lately that tap operator is usually the best used only for debugging purposes i.e to check memory leeks. you can use another map - also this observable is not completed as is not returning router.navigate

Reply:
Can you redirect me to a source regarding tap-debugging please? What should we use instead?

The `return` makes sense, I'll fix that.

Reply:
the source i have is in a book not with me.... i can explain tomorrow why is not a good idea... something to do with side effects... as if you resue the stream somewhere else i bring that side effect with it...